tests/Fixer/ClassUsage/DateTimeImmutableFixerTest.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Fixer\ClassUsage;
+
+use PhpCsFixer\Tests\Test\AbstractFixerTestCase;
+
+/**
+ * @author Kuba Werłos <werlos@gmail.com>
+ *
+ * @internal
+ *
+ * @covers \PhpCsFixer\Fixer\ClassUsage\DateTimeImmutableFixer
+ */
+final class DateTimeImmutableFixerTest extends AbstractFixerTestCase
+{
+    /**
+     * @param string      $expected
+     * @param null|string $input
+     *
+     * @dataProvider provideFixCases
+     */
+    public function testFix($expected, $input = null)
+    {
+        $this->doTest($expected, $input);
+    }
+
+    public function provideFixCases()
+    {
+        return [
+            [
+                '<?php new DateTimeImmutable();',
+                '<?php new DateTime();',
+            ],
+            [
+                '<?php new DateTimeImmutable();',
+                '<?php new DATETIME();',
+            ],
+            [
+                '<?php new \DateTimeImmutable();',
+                '<?php new \DateTime();',
+            ],
+            [
+                '<?php new Foo\DateTime();',
+            ],
+            [
+                '<?php namespace Foo; new DateTime();',
+            ],
+            [
+                '<?php namespace Foo; new \DateTimeImmutable();',
+                '<?php namespace Foo; new \DateTime();',
+            ],
+            [
+                '<?php namespace Foo; use DateTime; new \DateTimeImmutable();',
+                '<?php namespace Foo; use DateTime; new DateTime();',
+            ],
+            [
+                '<?php namespace Foo; use DateTime; new Bar\DateTime();',
+            ],
+            [
+                '<?php namespace Foo; use DateTime\Bar; new DateTime();',
+            ],
+            [
+                '<?php namespace Foo; use Bar\DateTime; new DateTime();',
+            ],
+            [
+                '<?php namespace Foo; use DateTime\Bar; use DateTime; use Baz\DateTime as BazDateTime; new \DateTimeImmutable();',
+                '<?php namespace Foo; use DateTime\Bar; use DateTime; use Baz\DateTime as BazDateTime; new DateTime();',
+            ],
+            [
+                '<?php $foo = DateTimeImmutable::ISO8601;',
+                '<?php $foo = DateTime::ISO8601;',
```

Comment:
Not sure this use case is relevant. IMO this should be ignored or optional.

Reply:
Done.